### PR TITLE
Add `apicache` on `/oracle/:ticker` route

### LINKED ISSUE #20: Speed up the oracle/:ticker endpoint
Looking at deployed oracles, it seems that the response time is around 4 seconds to get the attestation. it looks huge :thinking:  We could at least speed up the signing process by caching signatures (re-sign/re-fetch prices only when the timestamp value expires from a blocktime point of view)?

image

### PR DESCRIPTION
**Pull request by louisinger**
**This PR adds a route caching system on `oracle/:ticker`**

- [apicache](https://www.npmjs.com/package/apicache) keeps the signed message returned by `/oracle/:ticker` in memory during 30 seconds
- speed up the response time while the oracle is "spammed" (craft msg ~ 4000ms, returns cached value ~1ms)

it closes #20 

@tiero please review

### CODE REVIEW

File: src/utils/cache.ts
Comment (bordalix) [inline src/utils/cache.ts:1]: Do you need this empty file?

Comment (louisinger) [reply src/utils/cache.ts:1]: no, thanks :pray:

**Comment by easter-monolith** [comment]
With this PR the Oracle will have, on average, a 15 seconds delay to the real value.
Maybe I'm just being paranoid, but I fear this will allow some type of front running in periods of high volatility.

**Comment by easter-monolith** [comment]
@louisinger what if we have a continuous script generating static messages every 4 seconds (the time it takes to generate a new signed message) and the webserver just serve the static messages?

**Comment by louisinger** [comment]
> With this PR the Oracle will have, on average, a 15 seconds delay to the real value.
Maybe I'm just being paranoid, but I fear this will allow some type of front running in periods of high volatility.

right but most of the price endpoints are cached anyway. For instance coingecko free API cache all routes [from 1 to 5 minutes](https://support.coingecko.com/hc/en-us/articles/4538807536665-How-often-does-data-get-updated-or-refreshed-). So we probably have a bigger delay currently :thinking: 

> what if we have a continuous script generating static messages every 4 seconds (the time it takes to generate a new signed message) and the webserver just serve the static messages?

It has the disadvantage to run even if nobody is requesting the message (and so request the APIs which can be dangerous from rate limit POV). We could do the same thing by setting cache duration to 4s. However, note that this "4 seconds" is not a constant and depends on price sources performances/activity.

**Comment by tiero** [comment]
Concept ACK

**Pull request merged**